app/src/main/kotlin/compiler/ast/AstFactory.kt

```diff
@@ -74,7 +74,7 @@ object AstFactory {
     }
 
     private fun processConst(parseTree: ParseTree<Symbol>): Expression {
-        return when (parseTree.token()) {
+        return when ((parseTree as ParseTree.Branch).children[0].token()) {
             TokenType.INTEGER -> Expression.NumberLiteral((parseTree as ParseTree.Leaf).content.toInt())
```

Comment:
Here you also should have `.children[0]`

Reply:
Right. Corrected.